docs/principles/testing.md

```diff
@@ -0,0 +1,45 @@
+# Testing Strategy
+
+## End-to-End Testing
+
+End-to-end testing tests an entire product, service or application. Individual products should each have their own tests.
+
+### Cypress
```

Comment:
Headings here are a mix of tools and the things those tools are testing. Maybe they should all be in the form "Web sites are tested with Cypress", "APIs are tested with Jest", etc.?

Reply:
Good call, I've made a change for that.